### PR TITLE
Fix accuracy issue in minimax_m2 with TP > 1

### PR DESCRIPTION
**Pull request by skavulya**
Fix accuracy of minimax m2 for tensor parallel size > 1. Reduce is handled in FusedMoE after https://github.com/vllm-project/vllm-gaudi/pull/1377 and `reduce_results=False` dropped https://github.com/vllm-project/vllm-gaudi/pull/1444 

**Output without this PR:**
```
curl http://localhost:8000/v1/chat/completions     -H "Content-Type: application/json"     -d '{
        "model": "/mnt/weka/data/llm-d-models-pv/MiniMaxAI-MiniMax-M2.7",
        "messages": [
            {"role": "user", "content": [{"type": "text", "text": "Write a quick sort algorithm in python"}]}
        ], "max_tokens": 200
    }'
```
{"id":"chatcmpl-8eb68aec66d7f527","object":"chat.completion","created":1778891236,"prompt_routed_experts":null,"model":"/mnt/weka/data/llm-d-models-pv/MiniMaxAI-MiniMax-M2.7","choices":[{"index":0,"message":{"role":"assistant","content":"<think>I hadnet me find a programme2/apto/c- 241?._o.  no (the operation.yb-b\n> ыйо, not change this;~~ I think_colour ==\"light pink\";}) in...\n**The These must be not} was\n  and \n\n</think>):\n\nI('key=ельблиматš micrac_ / 1)2rasm_0.2 → add__2dict_eagle/tabString/im不过是 \\_list-ofchf_one \nCompute_with_prt_init: (New Tool Pro)\n-Main%-day__ ** [B1] : {nb_z0'];\n--own-traor:  with:   =: use  0.096-10_l_`this col0: 26;```\n</t_lN-蔓音频四文アنتストu+002:htt 도 원책임.(↑): The thought_dirty_s","refusal":null,"annotations":null,"audio":null,"function_call":null,"tool_calls":[],"reasoning":null},"logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"routed_experts":null}],"service_tier":null,"system_fingerprint":"vllm-0.20.1rc1.dev276+g54f548e9e-tp4-ep-614b7488","usage":{"prompt_tokens":45,"total_tokens":245,"completion_tokens":200,"prompt_tokens_details":null},"prompt_logprobs":null,"prompt_token_ids":null,"kv_transfer_params":null}

**With PR**

{"id":"chatcmpl-b79acb2e48acc5d0","object":"chat.completion","created":1778891747,"prompt_routed_experts":null,"model":"/mnt/weka/data/llm-d-models-pv/MiniMaxAI-MiniMax-M2.7","choices":[{"index":0,"message":{"role":"assistant","content":"<think>We are going to write a quick sort algorithm in Python.\n We will define a function quicksort that takes a list as input.\n We will choose a pivot (commonly the last element, but we can also choose a random element or the middle).\n We will partition the list into two parts: elements less than the pivot and elements greater than the pivot.\n Then we recursively sort the two parts and combine them with the pivot in between.\n\n However, note that the problem asks for a quick sort algorithm, so we'll implement the standard in-place quick sort.\n\n Steps:\n 1. If the list has length 0 or 1, it is already sorted.\n 2. Otherwise, select a pivot (we'll use the last element for simplicity).\n 3. Partition the list into two sublists: left (elements less than pivot) and right (elements greater than or equal to pivot).\n 4. Return the sorted left part, then the pivot, then the sorted right part.\n\n Alternatively, we","refusal":null,"annotations":null,"audio":null,"function_call":null,"tool_calls":[],"reasoning":null},"logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"routed_experts":null}],"service_tier":null,"system_fingerprint":"vllm-0.20.1rc1.dev276+g54f548e9e-tp4-ep-614b7488","usage":{"prompt_tokens":45,"total_tokens":245,"completion_tokens":200,"prompt_tokens_details":null},"prompt_logprobs":null,"prompt_token_ids":null,"kv_transfer_params":null}

### CODE REVIEW

File: vllm_gaudi/models/minimax_m2.py [hunk 1]
```diff
@@ -108,8 +108,6 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
         router_logits, _ = self.gate(hidden_states.to(torch.float32))
         final_hidden_states = self.experts(hidden_states=hidden_states, router_logits=router_logits)
```
[the remark this answers is not in the document]
Comment (skavulya) [reply vllm_gaudi/models/minimax_m2.py:109]: Reduce is already handled here https://github.com/vllm-project/vllm-gaudi/blame/f1abfec3e7ee4ecb7bb937624911ce112569ecca/vllm_gaudi/ops/hpu_fused_moe.py#L327

**Comment by mounikamandava** [comment]
Verified the fix and looks good to me

**Review by iboiko-habana — APPROVED** [review]
